post-til-private/src/main/java/no/difi/meldingsutveksling/ptp/PayloadInterceptor.java

```diff
@@ -0,0 +1,65 @@
+package no.difi.meldingsutveksling.ptp;
+
+import org.slf4j.Logger;
+import org.springframework.ws.client.WebServiceClientException;
+import org.springframework.ws.client.support.interceptor.ClientInterceptor;
+import org.springframework.ws.context.MessageContext;
+
+import javax.xml.transform.Source;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.stream.StreamResult;
+import java.io.StringWriter;
+import java.lang.invoke.MethodHandles;
+import java.util.function.Consumer;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+public class PayloadInterceptor implements ClientInterceptor {
+    private static final Logger logger = getLogger(MethodHandles.lookup().lookupClass().getName());
+    private final Consumer<String> callback;
+
+    public PayloadInterceptor(Consumer<String> payloadConsumer) {
+        this.callback = payloadConsumer;
+    }
+
+
+    @Override
+    public boolean handleRequest(MessageContext messageContext) throws WebServiceClientException {
+        return true;
+    }
+
+    @Override
+    public boolean handleResponse(MessageContext messageContext) throws WebServiceClientException {
+        Source payloadSource = messageContext.getResponse().getPayloadSource();
+        if(payloadSource != null) {
+            String payload = asXmlString(payloadSource);
+            callback.accept(payload);
+        }
+        return true;
+    }
+
+    private String asXmlString(Source source) {
+        StringWriter sw = new StringWriter();
+        try {
+            final Transformer transformer = TransformerFactory.newInstance().newTransformer();
+            StreamResult sr = new StreamResult(sw);
+            transformer.transform(source, sr);
+            return sw.toString();
+        } catch (TransformerException e) {
+            logger.error("Failed to marshall webservice response to XML string", e);
+        }
+        return "";
+    }
+
+    @Override
+    public boolean handleFault(MessageContext messageContext) throws WebServiceClientException {
+        return true;
+    }
+
+    @Override
+    public void afterCompletion(MessageContext messageContext, Exception ex) throws WebServiceClientException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the declaration of thrown exception 'org.springframework.ws.client.WebServiceClientException' which is a runtime exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)